src/resolution_functions/models/lookuptables.py

```diff
@@ -0,0 +1,143 @@
+
+    This allows non-Gaussian shapes to be produced. For smooth interpolation
+    and data efficiency, the x-axis is scaled in proportion to an approximated
+    standard deviation. This standard deviation is fitted to a polynomial
+    function of energy.
+
+    Parameters
+    ----------
+    model_data
+        The data associated with the model for a given version of a given instrument.
+
+    Attributes
+    ----------
+    input
+        The names of the columns in the ``omega_q`` array expected by all computation methods, i.e.
+        the names of the independent variables ([Q, w]) that the model models.
+    data_class
+        Reference to the `PolynomialModelData` type.
+    npz
+        The .npz file containing the model data
+    citation
+    """
+    input = ('energy_transfer',)
+
+    data_class: ClassVar[type[ScaledTabultedModelData]] = ScaledTabulatedModelData
+
+    def __init__(self, model_data: ScaledTabulatedModelData, **_):
+        from pathlib import Path
+
+        from numpy.polynomial import Polynomial
+        from scipy.interpolate import RegularGridInterpolator
+
+        super().__init__(model_data)
+        self.data = np.load(Path(INSTRUMENT_DATA_PATH) / model_data.npz)
+
+        self.polynomial = Polynomial(coef=self.data["coef"],
+                                     domain=self.data["domain"],
+                                                   window=self.data["window"])
+        self._interp = RegularGridInterpolator((self.data["energy_transfer"], self.data["kernel_energies"]),
+                                               self.data["table"],
+                                               method="linear",
+                                               bounds_error=False,
+                                               fill_value=0.)
+
+    def get_characteristics(self, omega_q: Float[np.ndarray, 'energy_transfer dimension=1']
+                            ) -> dict[str, Float[np.ndarray, 'sigma']]:
+        """
+        Computes the broadening width at each value of energy transfer (`omega_q`).
+
+        The model approximates the broadening using the Gaussian distribution, so the returned
+        widths are in the form of the standard deviation (sigma).
+
+        Parameters
+        ----------
+        omega_q
+            The energy transfer in meV at which to compute the width in sigma of the kernel.
+            This *must* be a ``sample`` x 1 2D array where ``sample`` is the number of energy
+            transfers.
+
+        Returns
+        -------
+        characteristics
+            The characteristics of the broadening function, i.e. the Gaussian width as sigma.
+        """
+        return {'sigma': self.polynomial(omega_q[:, 0])}
+
+    def get_kernel(self,
+                   omega_q: Float[np.ndarray, 'sample dimension=1'],
+                   mesh: Float[np.ndarray, 'mesh'],
+                   ) -> Float[np.ndarray, 'sample mesh']:
```

Comment:
```suggestion
    def get_kernel(self,
                   points: Float[np.ndarray, 'sample dimension=1'],
                   mesh: Float[np.ndarray, 'mesh'],
                   ) -> Float[np.ndarray, 'sample mesh']:
```

Also, could you add a docstring?